syntax = "proto3";

message BasicData {
    string symbol = 1;
    string instrument_id = 2;
    string timestamp = 3;
}

message QuoteData {
    BasicData basic = 1;      // 基础数据
    string ask_size = 2;      // 卖一档数量
    string ask_price = 3;     // 卖一价格
    string bid_size = 4;      // 买一档数量
    string bid_price = 5;     // 买一价格
}

message SnapshotData{
    BasicData basic = 1;      // 基础数据
    string open = 2;          // 开盘价
    string high = 3;          // 今日最高
    string low = 4;           // 今日最低
    string price = 5;         // 当前价格
    string pre_close = 6;     // 昨日收盘价
    string volume = 7;        // 成交量
    string change = 8;        // 涨跌幅
    string change_ratio = 9;  // 成交额
    string ask_size = 10;     // 卖一档数量
    string ask_price = 11;    // 卖一价格
    string bid_size = 12;     // 买一档数量
    string bid_price = 13;    // 买一价格
}